api/lib/source/polkadotV0-source.js

```diff
@@ -454,30 +455,43 @@ class polkadotAPI {
 
   async getInactiveDelegationsForDelegatorAddress(delegatorAddress) {
     const api = await this.getAPI()
-    let inactiveDelegations = []
 
     // We always use stash address to query delegations
     delegatorAddress = await this.getStashAddress(delegatorAddress)
 
     const stakingInfo = await api.query.staking.nominators(delegatorAddress)
-    const allDelegations =
-      stakingInfo && stakingInfo.toJSON() ? stakingInfo.toJSON().targets : []
-    allDelegations.forEach((nomination) => {
-      inactiveDelegations.push(
-        this.reducers.delegationReducer(
-          this.network,
-          { who: nomination, value: 0 }, // we don't know the value for inactive delegations
-          this.store.validators[nomination],
-          delegationEnum.INACTIVE
-        )
+    const nominees = JSON.parse(stakingInfo).targets
+    const allDelegations = await useInactives(api, delegatorAddress, nominees)
+    const inactiveDelegations = allDelegations.nomsWaiting.concat(
+      allDelegations.nomsChilled
+    )
+    return inactiveDelegations.map((nomination) => {
+      return this.reducers.delegationReducer(
+        this.network,
+        { who: nomination, value: 0 }, // we don't know the value for inactive delegations
+        this.store.validators[nomination],
+        delegationEnum.INACTIVE
       )
     })
-    return inactiveDelegations
   }
 
-  // TODO: find out how to get all undelegations in Polkadot
-  getUndelegationsForDelegatorAddress() {
-    return []
+  async getUndelegationsForDelegatorAddress() {
+    const api = await this.getAPI()
+
+    // We always use stash address to query delegations
+    delegatorAddress = await this.getStashAddress(delegatorAddress)
+
+    const stakingInfo = await api.query.staking.nominators(delegatorAddress)
+    const nominees = JSON.parse(stakingInfo).targets
+    const allDelegations = await useInactives(delegatorAddress, nominees)
+    return allDelegations.nomsInactive.map((nomination) => {
```

Comment:
are inactive delegations "undelegations"?

Reply:
yep, but only `chilled` and `inactive` nominations, the other are `waiting` (the incoming ones)